python_modules/dagster/dagster_tests/core_tests/test_mask_user_code_errors.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any
+
+import pytest
+from dagster import job, op
+from dagster._core.errors import DagsterUserCodeProcessError
+from dagster._core.test_utils import environ, instance_for_test
+from dagster._seven import get_current_datetime_in_utc
+
+from ..api_tests.utils import get_bar_repo_handle
+
+
+@pytest.fixture()
+def instance():
+    with instance_for_test() as instance:
+        yield instance
+
+
+class UserError(Exception):
+    def __init__(self):
+        super(UserError, self).__init__(
+            "This is an error which has some sensitive information! My password is hunter2"
+        )
+
+
+@pytest.fixture(scope="function")
+def enable_masking_user_code_errors() -> Any:
+    with environ({"DAGSTER_MASK_USER_CODE_ERRORS": "1"}):
+        yield
+
+
+def test_masking_op_execution(enable_masking_user_code_errors) -> Any:
+    @op
+    def throws_user_error(_):
+        raise UserError()
+
+    @job
+    def job_def():
+        throws_user_error()
+
+    result = job_def.execute_in_process(raise_on_error=False)
+    assert not result.success
+    assert not any("hunter2" in str(event) for event in result.all_events)
+    step_error = next(event for event in result.all_events if event.is_step_failure)
+    assert (
+        step_error.step_failure_data.error
+        and step_error.step_failure_data.error.cls_name == "DagsterMaskedUserCodeError"
+    )
```

Comment:
i think equally important as testing that the error doesn't include the UserError is testing that the log output still does include it - that could be quite tricky to get write